public/app/plugins/datasource/loki/querybuilder/components/LabelBrowserModal.tsx

```diff
@@ -38,20 +48,24 @@ export const LabelBrowserModal = (props: Props) => {
 
   return (
     <Modal isOpen={isOpen} title="Label browser" onDismiss={onClose}>
-      <LocalStorageValueProvider<string[]> storageKey={LAST_USED_LABELS_KEY} defaultValue={[]}>
-        {(lastUsedLabels, onLastUsedLabelsSave, onLastUsedLabelsDelete) => {
-          return (
-            <LokiLabelBrowser
-              languageProvider={languageProvider}
-              onChange={onChange}
-              lastUsedLabels={lastUsedLabels}
-              storeLastUsedLabels={onLastUsedLabelsSave}
-              deleteLastUsedLabels={onLastUsedLabelsDelete}
-              app={app}
-            />
-          );
-        }}
-      </LocalStorageValueProvider>
+      {!labelsLoaded && <LoadingPlaceholder text="Loading labels..." />}
+      {!hasLogLabels && <p>No labels found.</p>}
```

Comment:
```suggestion
      {labelsLoaded && !hasLogLabels && <p>No labels found.</p>}
```

Should this be this?

Reply:
@matyax something like this was my initial thought but i'm not a huge fan on nested ternaries.

Reply:
When you start seeing these piling up it's usually a better option:

![imagen](https://user-images.githubusercontent.com/1069378/204509802-87f18233-683c-45d4-a5af-7358128e2d61.png)

Reply:
Imo just using early returns could offer best readability here, even though you may need to repeat ` <Modal isOpen={isOpen} title="Label browser" onDismiss={onClose}>` lines.

e.g.:
```ts
if (!labelsLoaded) {
    return (
      <Modal isOpen={isOpen} title="Label browser" onDismiss={onClose}>
        <LoadingPlaceholder text="Loading labels..." />
      </Modal>
    );
  }
  if (!hasLogLabels) {
    return (
      <Modal isOpen={isOpen} title="Label browser" onDismiss={onClose}>
        <p>No labels found.</p>
      </Modal>
    );
  }
  return (
    <Modal isOpen={isOpen} title="Label browser" onDismiss={onClose}>
      <LocalStorageValueProvider<string[]> storageKey={LAST_USED_LABELS_KEY} defaultValue={[]}>
        {(lastUsedLabels, onLastUsedLabelsSave, onLastUsedLabelsDelete) => {
          return (
            <LokiLabelBrowser
              languageProvider={datasource.languageProvider}
              onChange={onChange}
              lastUsedLabels={lastUsedLabels}
              storeLastUsedLabels={onLastUsedLabelsSave}
              deleteLastUsedLabels={onLastUsedLabelsDelete}
              app={app}
            />
          );
        }}
      </LocalStorageValueProvider>
    </Modal>
  );
```

Reply:
That's also a good option. What I don't like about that is 1) mounting and unmounting a `Modal` component during a transition (loading -> loaded with or without labels) and 2) having to 3x `<Modal isOpen={isOpen} title="Label browser" onDismiss={onClose}>`.

Reply:
Then what about keeping modal, just changing the modal content: 

```
  const LogBrowserContent = () => {
    if (!labelsLoaded) {
      return <LoadingPlaceholder text="Loading labels..." />;
    }

    if (labelsLoaded && !hasLogLabels) {
      return <div>No labels found.</div>;
    }

    return (
      <LocalStorageValueProvider<string[]> storageKey={LAST_USED_LABELS_KEY} defaultValue={[]}>
        {(lastUsedLabels, onLastUsedLabelsSave, onLastUsedLabelsDelete) => {
          return (
            <LokiLabelBrowser
              languageProvider={datasource.languageProvider}
              onChange={onChange}
              lastUsedLabels={lastUsedLabels}
              storeLastUsedLabels={onLastUsedLabelsSave}
              deleteLastUsedLabels={onLastUsedLabelsDelete}
              app={app}
            />
          );
        }}
      </LocalStorageValueProvider>
    );
  };

  return (
    <Modal isOpen={isOpen} title="Label browser" onDismiss={onClose}>
      <LogBrowserContent />
    </Modal>
  );```